FIXED: Adding Total to Consumable View Page

akemidx opened the pull request:
This adds Total to the Consumable View page

OLD:
<img width="585" alt="Screenshot 2025-03-19 at 3 34 47 PM" src="https://github.com/user-attachments/assets/5d47b0a0-b16e-4f7c-9091-83076a49b320" />

NEW:
<img width="548" alt="Screenshot 2025-03-19 at 3 34 32 PM" src="https://github.com/user-attachments/assets/b3f18fe1-694f-4bac-9887-8e12ba84c557" />

akemidx committed:
adding Total to consumable view page

marcusmoore reviewed (approved):
This works but I wonder if we match the order from the index? The order in this PR, Total Remaining Min, makes the most sense to me and maybe we should update the index to follow that order as well?

![image](https://github.com/user-attachments/assets/26dbb923-38e8-4dde-86db-11e7963555f8)

Approving this code-wise but @snipe might have a some input to add.

akemidx committed:
data order

snipe committed:
Use develop branch

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16900 from marcusmoore/fixes/user-full-name-accessor

Handle settings not being available in full name accessor

marcusmoore committed:
Add test for existing functionality

marcusmoore committed:
Fix bug in getImageUrl method

marcusmoore committed:
Formatting

marcusmoore committed:
Add failing test

marcusmoore committed:
Handle creator being deleted

marcusmoore committed:
Remove label logo outline from L7162_B label

snipe committed:
Merge pull request #16920 from marcusmoore/fixes/remove-outline-from-label

Remove logo outline from L7162_B

marcusmoore committed:
Remove the replaced locales form macro

marcusmoore committed:
Remove unneeded div

snipe committed:
Null operator for missing created_by record

Signed-off-by: snipe <snipe@snipe.net>

marcusmoore committed:
Add failing tests

marcusmoore committed:
Improve method name

marcusmoore committed:
Invert method

marcusmoore committed:
Separate checking for sending email and webhook notifications

marcusmoore committed:
Formatting

marcusmoore committed:
Scaffold tests

marcusmoore committed:
Ensure CC emails are always sent for assets

snipe committed:
Merge branch 'develop' into bug/sc-29151

snipe committed:
Merge pull request #16918 from marcusmoore/bug/sc-29151

Handle displaying deleted creator of an accessory

uberbrady committed:
WIP: new tests for checkin/checkout counters

note that the test isnt going to work

More WIP for checkout counters

Got all tests passing

snipe committed:
Merge pull request #16921 from marcusmoore/chore/locale-select

Replace locales macro

snipe committed:
Merge pull request #16916 from marcusmoore/bug/sc-29153

Handle potential hard exception in Asset@getImageUrl method

snipe committed:
Add @JassonCordones as a contributor

snipe committed:
Fixed typo

Signed-off-by: snipe <snipe@snipe.net>

uberbrady committed:
A quick check to make sure that webhooks still fire when email is off

snipe committed:
Merge pull request #16942 from uberbrady/quick_temp_fix_notifications

A quick check to make sure that webhooks still fire when email is off

snipe committed:
Clearer text on status label types

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16943 from grokability/small-tweaks-to-status-label-text

Clearer text on status label types

marcusmoore committed:
Add failing test

marcusmoore committed:
Use route bound user instead of re-querying

marcusmoore committed:
Add assertion

marcusmoore committed:
Add comment

snipe committed:
Removed duplicate unique display in table

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Bumped hash

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use same logic for auditStore for validating unique fields

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Apply correct unique custom field validation for audit API

Signed-off-by: snipe <snipe@snipe.net>

marcusmoore committed:
Don't allow viewing edit page if user soft deleted

marcusmoore committed:
Merge branch 'develop' into fixes/webhook-checkin-checkout-fix

# Conflicts:
#	app/Listeners/CheckoutableListener.php

marcusmoore committed:
Strengthen test scenario

snipe committed:
Merge pull request #16946 from marcusmoore/bug/sc-29166

Gracefully handle error when editing of soft deleted users

marcusmoore committed:
Scaffold tests

marcusmoore committed:
Remove randomness from factory

marcusmoore committed:
Scaffold more tests

marcusmoore committed:
Improve test setup

marcusmoore committed:
Fix method name

marcusmoore committed:
WIP

marcusmoore committed:
Scaffold tests

marcusmoore committed:
WIP

marcusmoore committed:
WIP

marcusmoore committed:
Improve tests

marcusmoore committed:
WIP

marcusmoore committed:
WIP

marcusmoore committed:
Remove type hint

marcusmoore committed:
WIP

marcusmoore committed:
Formatting

marcusmoore committed:
Scaffold tests

marcusmoore committed:
WIP

marcusmoore committed:
WIP

marcusmoore committed:
Extract method

marcusmoore committed:
WIP

uberbrady committed:
Make checkin emails not send unless the send-emails attribute is set on the category

uberbrady committed:
Get rid of editorializing in comments

uberbrady committed:
Get rid of more editorialization

snipe committed:
Merge pull request #16953 from uberbrady/fix_checkin_emails

Make checkin emails not send when not configured to be

Godmartinz committed:
allow dynamic assignment to audit notifications

Godmartinz committed:
removes double scrollbar

snipe committed:
Merge pull request #16963 from Godmartinz/fix-double-scroll-bar-permission-groups

Removes double scrollbar from groups

marcusmoore committed:
Standardize test names

marcusmoore committed:
Merge branch 'develop' into fixes/webhook-checkin-checkout-fix

# Conflicts:
#	app/Listeners/CheckoutableListener.php

snipe committed:
Merge pull request #16962 from Godmartinz/audit_notifications_fix

Added dynamic properties to audit notifications

marcusmoore committed:
Bind the selected webhook and avoid clearing from database

marcusmoore committed:
Fix webhook selected value in mount method

marcusmoore committed:
Add migration to fix webhook settings

marcusmoore committed:
Check to make sure settings exist before attempting to update them

marcusmoore committed:
Fix assertion

marcusmoore committed:
Remove typehint

marcusmoore committed:
Improve method name

marcusmoore committed:
Improve method name

marcusmoore committed:
Improve display of user on accessory show page

marcusmoore committed:
Add translation

marcusmoore committed:
Do not show section if user doesn't exist

snipe committed:
Merge pull request #16932 from marcusmoore/fixes/webhook-checkin-checkout-fix

Improve notifications

marcusmoore committed:
Add authorization check

marcusmoore committed:
Extract variable

marcusmoore committed:
Add test

marcusmoore committed:
Add test

snipe committed:
Wider text fields

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Bumped hash

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16980 from uberbrady/add_new_checkin_checkout_counters_tests_rebased

New tests for checkin/checkout counters

snipe committed:
Merge pull request #16968 from marcusmoore/bug/sc-29233

Fixed potential slack webhook setting inconsistencies

snipe committed:
Merge pull request #16979 from marcusmoore/bug/sc-29178

Improved creator on accessory show page

snipe committed:
Bumped version

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Set audit warnings as int

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed #16958 - exclude archived assets from expiring assets report

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16982 from grokability/fixes#16958-expiring-assets-unarchived

Fixed #16958 - exclude archived assets from expiring assets report

snipe committed:
Moved faker out of dev reqs for seeding

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed test namespace

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16985 from grokability/move_faker_take_2

Moved faker out of dev reqs for seeding

snipe committed:
Manually add API headers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed noisy log

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
More small carbon fixes

Signed-off-by: snipe <snipe@snipe.net>

Tinyblargon committed:
fix: `PHP_UPLOAD_LIMIT` not set for PHP 8.3

snipe committed:
Improvements to API headers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16993 from Tinyblargon/fix-16992

fix: `PHP_UPLOAD_LIMIT` not set for PHP 8.3

snipe committed:
Add @Tinyblargon as a contributor

kovacs-andras committed:
Bumped container image versions

snipe committed:
Better handle missing asset payload because of RMB

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16999 from grokability/api_audt_fix

Better messaging when an asset fails validation on quick scan

snipe committed:
Refactorered limiting headers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed dupe semicolon

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Remove unused use statement after refactor

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Re-do the initial change :(

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added comment

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Extend the built-in ThrottleRequests middleware from Laravel

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated comments

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Add timestamp header

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed dupe header

Signed-off-by: snipe <snipe@snipe.net>

marcusmoore committed:
Avoid returning null from method that should return a boolean

snipe committed:
Merge pull request #17007 from marcusmoore/bug/sc-29278

Ensure boolean returned from method with boolean return type

marcusmoore committed:
Replace user_skin macro with blade component

marcusmoore committed:
Add semi-colon

marcusmoore committed:
Fix swapped yellow/yellow-dark

marcusmoore committed:
Adapt for regular "skin" macro

marcusmoore committed:
Improve component name

marcusmoore committed:
Remove skin macro

marcusmoore committed:
Inline two_factor_options macro

marcusmoore committed:
Inline customfield_elements select

marcusmoore committed:
Extract translation strings

marcusmoore committed:
Remove customfield_elements macro

snipe committed:
Merge pull request #16986 from grokability/api_throttle_headers

Fixed  #16961 - Manually add API headers

Robert-Azelis committed:
Update LocationsController.php - added company_id for API request

Robert-Azelis committed:
Update AssetModelsController.php - added requestable for API request

snipe committed:
Removed blade::render

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17014 from grokability/fix_breadcrumb_crash

Manually add API headers

snipe committed:
Removed extra a href

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17012 from Robert-Azelis/patch-8

API Locations - added company_id for API request

amedranogil committed:
more robust php.ini update.
could solve #10830 when setting PHP_UPLOAD_LIMIT environment variable in docker command and/or docker-compose.

snipe committed:
Updated languages

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17017 from grokability/localization/translations-2025-05-28

Updated translations

marcusmoore committed:
Scaffold test

snipe committed:
Added blade element

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added status label to view blade,  variable to controller method

Signed-off-by: snipe <snipe@snipe.net>

akemidx committed:
adding cookie for Assigned Assets

marcusmoore committed:
Scaffold test

marcusmoore committed:
Enforce min of .1 for label width and height

snipe committed:
Merge pull request #17020 from akemidx/column_persist_on_assigned_assets

FIXED: Column persist on User's Self View of Assigned Assets

marcusmoore committed:
Add validation

akemidx committed:
date picker

akemidx committed:
date picker

akemidx committed:
date picker

akemidx committed:
clearing the date pickers

akemidx committed:
clearing the date pickers

marcusmoore committed:
Reference correct translation string

akemidx committed:
fixing two translation strings.

snipe committed:
Merge pull request #17022 from marcusmoore/bug/sc-29281

Avoid dividing by zero in DefaultLabel

Robert-Azelis committed:
Update AssetModelsController.php

snipe committed:
Fixed route for custom fields index

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17027 from akemidx/bug/sc-29295

FIXED: Translation strings in Username/Email formats

snipe committed:
Merge pull request #17026 from marcusmoore/bug/translate-email-format

Reference correct translation string

snipe committed:
Merge pull request #17024 from akemidx/created_at_date_picker

ADDED: Created At date picker on Custom Reports

snipe committed:
Merge pull request #17025 from akemidx/clear_button_on_date_picker

FIXED: Clear Button actually Clearing Dates on Date Picker

snipe committed:
Merge pull request #17010 from marcusmoore/chore/replace-two-factor-select-macro

Replace `two_factor_options` macro

snipe committed:
Merge pull request #17009 from marcusmoore/chore/replace-user-skin-macro

Replace `skin` and `user_skin` macros with blade component

snipe committed:
Fixed #17034 - larger header color box on small views

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17035 from grokability/settings_style_improvements

Fixed #17034 - larger header color box on small views

snipe committed:
Added supplier importer

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added icon

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Set correct redirect

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed mapping

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added locations and supplier import types

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added strings

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added tests and test support

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added columns to suppliers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added sample import CSVs

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17037 from grokability/supplier_importer

Added #17036 - suppliers importer

snipe committed:
Use presenter for suppliers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added created_by to location presenter

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added adminuser method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added created_by to transformer

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Eager load adminuser method

Signed-off-by: snipe <snipe@snipe.net>

marcusmoore committed:
Avoid displaying empty table

snipe committed:
Added additional search and display fields for suppliers and locations

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17040 from marcusmoore/improve-acceptance-reminder-output

Avoid displaying empty table in `SendAcceptanceReminder` command

snipe committed:
Removed unused method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed stray character

Typing is hard :(

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17041 from grokability/improve_locations_and_supplier_api

Small refinements for suppliers and locations API and list view

marcusmoore committed:
Add failing test

marcusmoore committed:
Handle asset not having location

marcusmoore committed:
Avoid dumping pdf contents to test results

Commit:
Docker: Ensure permissions on Laravel log file

FIXES: https://github.com/grokability/snipe-it/issues/12725

In some of our Docker startups, it was possible for the Laravel log file
to be created with root permissions, causing future errors when the
non-root webapp tries to write to it.

We'll now always chown (and create, if necessary) the log file to the proper
user after running any artisan commands (as root)

We _could_ run them as the proper user via su, but IMO not doing so keeps the
script easier to read, but I'm not married to the approach. I'd still
want to keep the chown command(s) in, because it will also fix the
permissions for anyone who already has this issue.

snipe committed:
Merge pull request #17045 from grokability/docker-laravel-log-permissions

Docker: Ensure permissions on Laravel log file

snipe committed:
Merge pull request #17044 from marcusmoore/bug/sc-29302

Handle missing location when rendering labels

snipe committed:
Added manufacturer icon

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added redirect after import

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Skip manufacturers on checkout import type

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added manufacturer importer

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added manufacturers import fields

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added manufacturer factory

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed “send welcome email” detection

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added sample CSV

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added tests and support helper

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17061 from grokability/add_manufacturer_importer

Added manufacturer importer

snipe committed:
Added category importer

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed variable name

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17042 from marcusmoore/chore/replace-customfield-elements-macro-take-two

Replace customfield_elements form macro take two

snipe committed:
Fixed weird layout

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Improved CSV

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17062 from grokability/add_category_importer

Added category importer

snipe committed:
Merge pull request #17019 from grokability/#15320-status_to_bulk_checkout

Fixed #15320 - added status label to bulk checkout

snipe committed:
Merge pull request #17013 from Robert-Azelis/patch-9

API Models - added requestable for API request

snipe committed:
Merge pull request #17038 from amedranogil/develop

more robust php.ini update.

snipe committed:
Add @amedranogil as a contributor

uberbrady committed:
Fix to rate-limiter on higher rate-limits

snipe committed:
Merge pull request #17074 from uberbrady/improve_api_rate_limiting

Fix to rate-limiter on higher rate-limits

snipe committed:
Added over sixty test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Duplicates PR #16957

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Bumped version

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Disable form fields if the value is blank

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17078 from grokability/fixes-#17076-validation-on-bulk-submit

Fixed #17076 - Disable optional status ID form field if value is blank/Do Not Change

snipe committed:
Updated tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added action_date

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed action date from array of things to log

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Null both assigned to and assigned type on delete

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added console command for fixup

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added time on action date

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added migration to fix existing wonky data

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17087 from grokability/#17085-checkin-and-delete-not-nulling-assigned_type

Fixed #17085 - assigned_type not being nulled on asset delete+checkin

snipe committed:
Removed log error

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #16998 from kovacs-andras/develop

Bumped container image versions

snipe committed:
Make version number match tagged version :(

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Remove error log

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed #16240 - made additional strings translatable

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Remove unused translation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added aria tag for accessibility

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17099 from grokability/fixes-16240-localization-strings

Fixed #16240 - made additional strings translatable

uberbrady committed:
Add deleted_at index to action_logs for people with many deleted action_logs

snipe committed:
Added advanced search to users

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17101 from grokability/fixes-16157-added-advanced-search-to

Fixed #16157 - Added advanced search to users

snipe committed:
Merge pull request #17100 from uberbrady/add_deleted_at_index_to_action_logs

Fixed #16205 - Add an index to deleted_at for those with many deleted action_logs

snipe committed:
Fixed #17102 - added keywords to admin settings search for notifications

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17103 from grokability/fixes-17102-add-cc-as-search-string-in-settings

Fixed #17102 - added keywords to admin settings search for notifications

snipe committed:
Fixed #16218 - added centos to the switch case

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #17107 from grokability/fixes-16218-add-centos

Fixed #16218 - added centos to the switch case

akemidx committed:
Merge remote-tracking branch 'origin/bug/sc-28718' into bug/sc-28718
